structure_tests/run.sh

```diff
@@ -74,4 +76,8 @@ if [ -z "$IMAGE_NAME" ]; then
 	usage
 fi
 
-docker run --privileged=true -v /workspace:/workspace --entrypoint="$ENTRYPOINT" "$IMAGE_NAME" -c "$CMD_STRING"
+# Get id of container we're currently in.  This method is
+# system-specific but probably better than looking at $HOSTNAME.
+THIS_CONTAINER=$(basename $(head -1 /proc/self/cgroup))
```

Comment:
It's a bit far-fetched, but you could do `docker run --name=foo --hostname=bar` which would mess up just looking at `$HOSTNAME`.  Also the hostname gets truncated at 12 characters so [collisions](https://github.com/docker/docker/issues/28260) can happen.

`THIS_CONTAINER=$HOSTNAME` is a whole lot simpler though.